### PR TITLE
Avr smart shooting

### PR DESCRIPTION
**Pull request by dKarangula**

### CODE REVIEW

File: src/main/java/frc/robot/commands/OuttakeManual/OuttakeManual.java [hunk 1]
```diff
@@ -32,23 +42,52 @@ public void execute() {
...
-        double botAngleToSpeaker =
-            FiringSolutionHelper.getVectorToSpeaker(
-                    RobotCoordinator.getInstance().getRobotXPos(),
-                    RobotCoordinator.getInstance().getRobotYPos())
-                .getAngle()
-                .getDegrees();
-        solution =
-            FiringSolutionManager.getInstance().calcSolution(botMagToSpeaker, botAngleToSpeaker);
+        // Only calculate new firing solutions if we can see the target April tag with Limelight
+        if (outtakeLimelight.getTargetVisible()) {
```
Comment (zaydkhan4322) [inline src/main/java/frc/robot/commands/OuttakeManual/OuttakeManual.java]: this needs to check if we're seeing the correct tag, not just any target

File: src/main/java/frc/robot/commands/OuttakeManual/OuttakeManual.java [hunk 2]
```diff
@@ -32,23 +42,52 @@ public void execute() {
...
+
+          final Pose2d botPoseToSpeaker;
+          if (Robot.isRed()) {
+            botPoseToSpeaker = outtakeLimelight.getTargetPose3DToBot(Constants.FieldConstants.redSpeakerCenterTagID).toPose2d();
+          }
+          else {
+            botPoseToSpeaker = outtakeLimelight.getTargetPose3DToBot(Constants.FieldConstants.blueSpeakerCenterTagID).toPose2d();
+          }
+          
+          // Round to two decimals to reduce noise and have a stable shooting angle and flywheel speed
```
Comment (zaydkhan4322) [inline src/main/java/frc/robot/commands/OuttakeManual/OuttakeManual.java]: i'd round to 1 decimal, if not just a whole number. we can also consider into rounding by .2s or something (mult/divide by 5) if we want

**Review by zaydkhan4322 — CHANGES_REQUESTED** [review]
please see targeting and rounding change

File: src/main/java/frc/robot/commands/OuttakeManual/OuttakeManual.java (hunk 2 above)
Comment (dKarangula) [reply src/main/java/frc/robot/commands/OuttakeManual/OuttakeManual.java]: Is rounding to whole number or 1 decimal precise enough with units as meters? The farther we go from the speaker, the more accurate we need our shooting interpolation. Our current interpolation has multiple shots for 1 meter, and I'm not sure if rounding to 1 or 0 decimals is accurate enough.

File: src/main/java/frc/robot/RobotContainer.java [hunk 3]
```diff
@@ -295,10 +295,9 @@ private void configureButtonBindings() {
       operatorXbox
           .y()
           .onTrue(
-              Commands.runOnce(
-                  () ->
-                      outtakeManual.updateStateMachine(
-                          OuttakeManualTrigger.ENABLE_SMART_SHOOTING)));
+              Commands.runOnce(() -> {
+                outtakeManual.setFiringSolution(Constants.FiringSolutions.DefaultSmartShooting);
```
Comment (zaydkhan4322) [inline src/main/java/frc/robot/RobotContainer.java:299]: this implementation is not what I was thinking of but it's fine

**Review by zaydkhan4322 — APPROVED** [review]

File: src/main/java/frc/robot/RobotContainer.java (hunk 3 above)
Comment (zaydkhan4322) [reply src/main/java/frc/robot/RobotContainer.java:299]: we can talk about it at the meeting if you want

**Pull request merged**